nft_ingester/src/index_syncronizer.rs

```diff
@@ -74,38 +77,60 @@ where
         rx: &tokio::sync::broadcast::Receiver<()>,
         run_full_sync_threshold: i64,
     ) {
-        if let Ok(SyncStatus::FullSyncRequired(_)) =
-            self.get_sync_state(run_full_sync_threshold).await
-        {
-            let res = self.full_syncronize(rx).await;
-            match res {
-                Ok(_) => {
-                    tracing::info!("Full synchronization finished successfully");
-                }
-                Err(e) => {
-                    tracing::error!("Full synchronization failed: {:?}", e);
+        let asset_types = [AssetType::Fungible, AssetType::NonFungible];
+        for asset_type in asset_types {
+            if let Ok(SyncStatus::FullSyncRequired(_)) = self
+                .get_sync_state(run_full_sync_threshold, asset_type)
+                .await
+            {
+                let res = self.full_syncronize(rx).await;
+                match res {
+                    Ok(_) => {
+                        tracing::info!("Full synchronization finished successfully");
+                    }
+                    Err(e) => {
+                        tracing::error!("Full synchronization failed: {:?}", e);
+                    }
                 }
             }
         }
     }
 
-    pub async fn run(
+    pub async fn non_fungible_run(
         &self,
         rx: &tokio::sync::broadcast::Receiver<()>,
         run_full_sync_threshold: i64,
         timeout_duration: tokio::time::Duration,
     ) {
         while rx.is_empty() {
-            let res = self
-                .synchronize_asset_indexes(rx, run_full_sync_threshold)
-                .await;
-            match res {
-                Ok(_) => {
-                    tracing::info!("Synchronization finished successfully");
-                }
-                Err(e) => {
-                    tracing::error!("Synchronization failed: {:?}", e);
-                }
+            if let Err(e) = self
+                .synchronize_non_fungible_asset_indexes(rx, run_full_sync_threshold)
+                .await
+            {
+                tracing::error!("Synchronization failed: {:?}", e);
+            } else {
+                tracing::info!("Synchronization finished successfully");
```

Comment:
messaging should indicate which one is that

Reply:
Fixed